lib/Config.ts

```diff
@@ -164,37 +169,44 @@ class Config {
     }
   }
 
-  /**
-   * Updates the default values for all fields derived from the xu directory when a custom
-   * xu directory is specified by the config file or command line arguments.
-   */
-  private updateDefaultPaths = (xudir: string) => {
-    // if we have a custom xu directory, update the default values for all fields that are
-    // derived from the xu directory.
-    this.xudir = xudir;
-    this.logpath = this.getDefaultLogPath();
-    this.dbpath = this.getDefaultDbPath();
+  private getNetwork = (args: { [argName: string]: any }) => {
+    const networks: { [val: string]: boolean } = {
+      [XuNetwork.MainNet]: args.mainnet,
+      [XuNetwork.TestNet]: args.testnet,
+      [XuNetwork.SimNet]: args.simnet,
+      [XuNetwork.RegTest]: args.regtest,
+    };
+
+    const selected = Object.keys(networks).filter(key => networks[key]);
+    if (selected.length > 1) {
+      throw Error('only one network selection is allowed');
+    }
+
+    if (selected.length === 0) {
+      return XuNetwork.SimNet;
+    } else {
+      return selected[0] as XuNetwork;
+    }
   }
 
-  private updateMacaroonPaths = (network: string) => {
-    this.network = network as Network;
+  private updateMacaroonPaths = () => {
     for (const currency in this.lnd) {
       switch (currency) {
         case 'LTC':
           // litecoin uses a specific folder name for testnet
           this.lnd.LTC!.macaroonpath = path.join(this.lnd.LTC!.macaroonpath, '..', '..',
-            this.network === Network.TestNet ? 'testnet4' : this.network, 'admin.macaroon');
+            this.network === XuNetwork.TestNet ? 'testnet4' : this.network, 'admin.macaroon');
           break;
         default:
           // by default we want to update the network folder name to the selected network
-          this.lnd[currency]!.macaroonpath = path.join(this.lnd[currency]!.macaroonpath, '..', '..', network, 'admin.macaroon');
+          this.lnd[currency]!.macaroonpath = path.join(this.lnd[currency]!.macaroonpath, '..', '..', this.network, 'admin.macaroon');
           break;
       }
     }
   }
 
   private getDefaultDbPath = () => {
-    return path.join(this.xudir, 'xud.db');
+    return path.join(this.xudir, `xud-${this.network}.db`);
```

Comment:
Just a thought but we can keep the `xud.db` filename for mainnet and only specify a network if it's not mainnet, so this would be something like:

```
this.network === XuNetwork.MainNet ? 'xud.db' : `xud-${this.network}.db`
```

Reply:
I wouldn't mind it being more explicit for each network.

Reply:
Yes, i'd rather have it be more explicit.